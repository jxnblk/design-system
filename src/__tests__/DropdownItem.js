import React from 'react'
import renderer from 'react-test-renderer'
import { DropdownItem, theme } from '..'

describe('DropdownItem', () => {
  test('renders with itemIdenx prop', () => {
    const json = renderer
      .create(
        <DropdownItem itemIndex={1} isSelected>
          Test 1
        </DropdownItem>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('list-style-type', 'none')
  })

  test('renders with itemIndex prop', () => {
    const json = renderer
      .create(<DropdownItem itemIndex={8}>Test 2</DropdownItem>)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('list-style-type', 'none')
  })

  test('renders with role prop', () => {
    const json = renderer
      .create(<DropdownItem role="menuitem">Test 2</DropdownItem>)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('list-style-type', 'none')
  })

  test('renders with tabIndex prop', () => {
    const json = renderer
      .create(<DropdownItem tabIndex={-1}>Test 2</DropdownItem>)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('list-style-type', 'none')
  })

  test('renders with isSelected prop', () => {
    const json = renderer
      .create(<DropdownItem isSelected>Test 2</DropdownItem>)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('list-style-type', 'none')
    expect(json).toHaveStyleRule('background-color', theme.colors['blue'])
    expect(json).toHaveStyleRule('color', theme.colors['white'])
  })
})
