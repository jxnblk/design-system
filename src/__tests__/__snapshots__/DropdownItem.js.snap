// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem renders with isSelected prop 1`] = `
.c0 {
  list-style-type: none;
  background-color: #007aff;
  color: #fff;
}

.c0:hover {
  background-color: #007aff;
  color: #fff;
}

<li
  className="c0"
  id={null}
  role={undefined}
  tabIndex={undefined}
>
  Test 2
</li>
`;

exports[`DropdownItem renders with itemIdenx prop 1`] = `
.c0 {
  list-style-type: none;
  background-color: #007aff;
  color: #fff;
}

.c0:hover {
  background-color: #007aff;
  color: #fff;
}

<li
  className="c0"
  id="mi1"
  role={undefined}
  tabIndex={undefined}
>
  Test 1
</li>
`;

exports[`DropdownItem renders with itemIndex prop 1`] = `
.c0 {
  list-style-type: none;
}

.c0:hover {
  background-color: #007aff;
  color: #fff;
}

<li
  className="c0"
  id="mi8"
  role={undefined}
  tabIndex={undefined}
>
  Test 2
</li>
`;

exports[`DropdownItem renders with role prop 1`] = `
.c0 {
  list-style-type: none;
}

.c0:hover {
  background-color: #007aff;
  color: #fff;
}

<li
  className="c0"
  id={null}
  role="menuitem"
  tabIndex={undefined}
>
  Test 2
</li>
`;

exports[`DropdownItem renders with tabIndex prop 1`] = `
.c0 {
  list-style-type: none;
}

.c0:hover {
  background-color: #007aff;
  color: #fff;
}

<li
  className="c0"
  id={null}
  role={undefined}
  tabIndex={-1}
>
  Test 2
</li>
`;
